plugins/Morpheus/javascripts/piwikHelper.js

```diff
@@ -33,7 +33,17 @@ function _pk_externalRawLink(url, values) {
   const sourceOverride = (typeof values != 'undefined' && values.length > 1 && values[1] ? values[1] : null);
   const mediumOverride = (typeof values != 'undefined' && values.length > 2 && values[2] ? values[2] : null);
 
-  const returnURL = new URL(url);
+  var returnURL= null;
+  try {
+      returnURL = new URL(url);
+  } catch(error) {
+      console.log('Error parsing URL: ' + url);
+      return '';
+  }
+  if (!returnURL) {
+    return '';
+  }
```

Comment:
We already use `let` and `const` so we can stick with `let` here, plus the early return from the catch is not much earlier than the null check right after that.

```suggestion
  let returnURL = null;
  try {
      returnURL = new URL(url);
  } catch(error) {
      console.log('Error parsing URL: ' + url);
  }
  if (!returnURL) {
    return '';
  }
```